File: leet-code/toNumberSum.go
package main

func toNumberSum(nums []int, target int) []int {
	numberMap := make(map[int]int)
	result := []int{}
	for index, value := range nums {
		remainder := target - value
		mValue, ok := numberMap[remainder]
		if ok {
			result = append(result, index)
			result = append(result, mValue)
			return result
		}
		numberMap[value] = index
	}
	return result
}

File: leet-code/toNumberSum.js

/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var toNumberSum = function (nums, target) {
  if (nums.length === 0) {
    return []
  }
  const map = new Map()

  for (let index = 0; index < nums.length; index++) {
    const value = nums[index]
    const remainder = target - value
    if (map.has(remainder)) {
      return [index, map.get(remainder)]
    }
    map.set(value, index)
  }
  throw new Error('no two number')
}

console.log(twoSum([3, 2, 4], 6))
